enigma-tools-m/src/primitives/km_primitives.rs

```diff
@@ -6,37 +10,52 @@ use crate::serde_json;
 use enigma_crypto::{rand, symmetric, CryptoError, Encryption, hash};
 use enigma_types::{ContractAddress, DhKey, PubKey, StateKey};
 
+/// A Message ID type, used to identify each message to the response.
 pub type MsgID = [u8; 12];
 
+/// An enum used to differentiate between `Request` and `Response`.
+/// and between Response before and after encryption
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
 #[serde(crate = "crate::serde")]
 pub enum PrincipalMessageType {
+    /// A Response from the KM node, containing a list of (Address, Key) tuples.
     Response(Vec<(ContractAddress, StateKey)>),
+    /// A Request for the KM node, with an optional list of addresses.
     Request(Option<Vec<ContractAddress>>),
+    /// The same as `Response` but this is after encryption.
     EncryptedResponse(Vec<u8>),
 }
 
+/// The Message struct used to communicate between a worker and the Key Management Node.
+/// this struct contains the data(request/response) the DH pubkey and the MsgID.
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
 #[serde(crate = "crate::serde")]
 pub struct PrincipalMessage {
+    /// The data of the message.
+    ///
+    /// This can be either Request/Response or an Encrypted Response.
     pub data: PrincipalMessageType,
     pub(crate) pubkey: Vec<u8>,
     pub(crate) id: MsgID,
 }
 
 impl PrincipalMessage {
+    /// This will create a new Message with a random MsgID.
     pub fn new(data: PrincipalMessageType, pubkey: PubKey) -> Result<Self, CryptoError> {
         let mut id = [0u8; 12];
         rand::random(&mut id)?;
         let pubkey = pubkey.to_vec();
         Ok(Self { data, pubkey, id })
     }
 
+    /// This should be used only by the KeyManagement node to create a response that will contain the same ID
+    /// as the request.
     pub fn new_id(data: PrincipalMessageType, id: [u8; 12], pubkey: PubKey) -> Self {
```

Comment:
this is a confusing name for the function, isn't it?

Reply:
in a different PR we can change it to `new_with_id` or something like that